### ISSUE TITLE
Button Captions for Screen Readers (Accessibility)

### ISSUE THREAD

**Issue by jesse-troy** [zulip/zulip-mobile#4232]
My organization is currently rolling out this platform, and it appears to be fulfilling all of our needs accept for one small issue.

For the visually impaired mobile users (specifically ios) , there appears to be a lack of pronounceable widgets in the user interface. It has been pointed out to me that the most likely buttons without this "caption" are the buttons to send a message and the button to back out of a conversation. There are other buttons without labels, but these are obviously the important ones. 

The fix for this was described to be similar to adding alt-text to images in HTML.

Would it be possible to add better usability for the visually impaired?

**Comment by timabbott** [comment]
@jesse-troy accessibility is certainly something we're working on for the project as a whole (a lot of work is going into accessibility for the visually impaired in the webapp right now).  I don't have specific iOS expertise, but I expect folks here would be happy to collaborate on identifying buttons missing accessibility labels and adding them.  Might be best to discuss details in chat.zulip.org, perhaps starting with discussion of how you're testing which buttons are missing this feature?

**Comment by gnprice** [comment]
Thanks @jesse-troy for raising this! I'd be glad to see someone pick this up and do it.

There was a previous PR #2753 a couple of years ago which made some good underlying changes which were merged, but then unfortunately the PR didn't get any further work after the initial review. The strategy I'd recommend is to look at what's in that PR branch and the comments on it and start from there.

Based on https://github.com/zulip/zulip-mobile/pull/2753#issuecomment-402915122 , which was the last comment I made reviewing that PR back in 2018, most of the commits in it were actually 100% good already -- they just depended on some earlier commits that needed changes, in particular the commit "a11y: Add `textWithUnreadCount` util function" as discussed in https://github.com/zulip/zulip-mobile/pull/2753#discussion_r200529450 . So it may not take a great deal of work for someone to get the whole thing into shape.

(I said much the same things [in a chat thread](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Screen.20Reader.20Accessibility/near/993131) the day this was filed, but neglected to copy here to GitHub so it's visible on this thread.)

**Comment by pratyaksh123** [comment]
Hey, I'd like to take this up.

**Comment by Gautam-Arora24** [comment]
Hey @gnprice ! Can I work on this issue?

**Comment by pratyaksh123** [comment]
As I can see the problem in that PR is with the `textWithUnreadCount` util function in https://github.com/zulip/zulip-mobile/pull/2753 and the solution is to use a ```<FormattedPlural/>``` (https://formatjs.io/docs/react-intl/components/#formattedplural) in that util.
Plus there are a lot of merge conflicts in that PR branch as the current codebase is updated quite a lot. So maybe we can start building a fresh PR taking help from that PR branch.
@Gautam-Arora24 maybe let's collaborate for this one!

**Comment by Gautam-Arora24** [comment]
@zulipbot claim

**Comment by gnprice** [comment]
@jesse-troy Thanks again for reporting this issue!

We've just added captions for those two particularly important buttons you mentioned. There's plenty more for us to do in this area; I've just filed #4394 as a general tracking issue for the broader project of adding captions to other buttons in the app, and making other improvements for using with a screen-reader. If you, or one of your colleagues who's visually impaired, would like to add feedback there on what's most important after these first two buttons, we'd be glad to hear more!
